Root out obsolete self-hosted setup – WEB-194

AlexTugarev opened the pull request:
This PR removes everything that was made for the old self-hosted setup page. 

## Related Issue(s)
<!-- List the issue(s) this PR solves -->
Fixes WEB-194

## How to test
<!-- Provide steps to test this PR -->

## Documentation
<!--
Does this PR require updates to the documentation at www.gitpod.io/docs?
* Yes
  * 1. Please create a docs issue: https://github.com/gitpod-io/website/issues/new?labels=documentation&template=DOCS-NEW-FEATURE.yml&title=%5BDocs+-+New+Feature%5D%3A+%3Cyour+feature+name+here%3E
  * 2. Paste the link to the docs issue below this comment
* No
  * Are you sure? If so, nothing to do here.
-->

#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - rm-sh-setup</li>
	<li><b>🔗 URL</b> - <a href="https://rm-sh-setup.preview.gitpod-dev.com/workspaces" target="_blank">rm-sh-setup.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
</ul>

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [x] with-ws-manager-mk2
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`
</details>
 
/hold

AlexTugarev commented on components/server/src/installation-admin/telemetry-data-provider.ts:
```diff
@@ -21,7 +21,6 @@ export class InstallationAdminTelemetryDataProvider {
         try {
             const data: TelemetryData = {
                 installationAdmin: await this.installationAdminDb.getData(),
-                totalUsers: await this.userDb.getUserCount(true),
```
Heads up! I consider the installation admin be not used, thus removing the `getUserCount` usage here seems to be save.

easyCZ commented on components/gitpod-db/src/user-db.ts:
```diff
@@ -32,13 +32,6 @@ export interface UserDB extends OAuthUserRepository, OAuthTokenRepository {
-     *
-     * @param excludeBuiltinUsers substract the builtin-users from the count (currently only the user builtin-workspace-prober), true by default
-     */
-    getUserCount(excludeBuiltinUsers?: boolean): Promise<number>;
```
This is being used in https://github.com/gitpod-io/gitpod/pull/17718/files#diff-315103e87ae1b7ee06fb5f3f99090d11a3748704f3799561ef3dd49d73ea0e33R18 AFAIK

AlexTugarev replied:
Thanks for pointing that out. 
I'll revert.

easyCZ commented on install/installer/pkg/config/v1/experimental/experimental.go:
```diff
@@ -265,7 +265,6 @@ type ServerConfig struct {
 	DisableWorkspaceGarbageCollection bool              `json:"disableWorkspaceGarbageCollection"`
 	DisableCompleteSnapshotJob        bool              `json:"disableCompleteSnapshotJob"`
 	InactivityPeriodForReposInDays    *int              `json:"inactivityPeriodForReposInDays"`
-	ShowSetupModal                    *bool             `json:"showSetupModal"`
```
Worth checking this is not configured in dedicated installations. If it is, we need to remove from there first.

AlexTugarev replied:
Going to revert this

svenefftinge replied:
Would be great to remove this config as well. I believe it is only used in cells that we do not intend to update?

easyCZ replied:
I think the proper fix should for dedicated to ignore unknown config values (rather than fail) as it's gonna become impossible to keep track of this

AlexTugarev replied:
change reverted.

AlexTugarev commented:
@easyCZ, I updated the branch and reverted the critical `getUserCount`. Let that be a problem of another day.
Would you mind to take another look and help land it.

/unhold
